apps/babydegen-ui/src/components/WithdrawAgentsFunds/useWithdrawFunds.ts

```diff
@@ -0,0 +1,85 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+import { Address } from '@agent-ui-monorepo/util-constants-and-types';
+
+import { WithdrawalStatus, WithdrawalInitiateResponse } from '../../types';
+import {
+  mockWithdrawInitiateResponse,
+  mockWithdrawStatusResponse,
+} from '../../mocks/mockFundsWithdrawal';
+import { LOCAL } from '../../constants/urls';
+
+const IS_MOCK_ENABLED = process.env.IS_MOCK_ENABLED === 'true';
+
+const initiateWithdrawal = async (targetAddress: Address): Promise<WithdrawalInitiateResponse> => {
+  if (IS_MOCK_ENABLED) {
+    return new Promise<WithdrawalInitiateResponse>((resolve) =>
+      setTimeout(() => resolve(mockWithdrawInitiateResponse), 2000),
+    );
+  }
+
+  const response = await fetch(`${LOCAL}/withdrawal/initiate`, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ target_address: targetAddress }),
+  });
+
+  if (!response.ok) throw new Error('Failed to initiate withdrawal.');
+  return response.json();
+};
+
+export const useWithdrawFunds = () => {
+  const [withdrawId, setWithdrawId] = useState<string | null>(null);
+
+  const { isPending, isError, mutateAsync } = useMutation<
+    WithdrawalInitiateResponse,
+    unknown,
+    Address
+  >({
+    mutationKey: ['withdraw-initiate'],
```

Comment:
```suggestion
    mutationKey: ['withdrawInitiate'],
```

I think we use camelCase for query keys